Tools/Scripts/libraries/webkitbugspy/webkitbugspy/bugzilla.py

```diff
@@ -461,6 +468,49 @@ def add_comment(self, issue, text):
 
         return result
 
+    def relate(self, issue, depends_on=None, blocks=None, regressed_by=None, regressions=None, **relations):
+        if relations:
+            raise TypeError("'{}' is an invalid relation".format(list(relations.keys())[0]))
+
+        update_dict = dict()
+        update_dict['ids'] = [issue.id]
+        if depends_on:
+            update_dict['depends_on'] = {'add': depends_on}
+        if blocks:
+            update_dict['blocks'] = {'add': blocks}
+        if regressed_by:
+            update_dict['regressed_by'] = {'add': regressed_by}
+        if regressions:
+            update_dict['regressions'] = {'add': regressions}
+
+        response = None
+        try:
+            response = self.session.put(
+                '{}/rest/bug/{}{}'.format(self.url, issue.id, self._login_arguments(required=True)),
+                json=update_dict,
+                timeout=self.timeout,
+            )
+        except RuntimeError as e:
```

Comment:
Does `self.session.puts`'s exception inherit from `RuntimeError`?

Reply:
To be honest, I took the error handling from existing code in `set`. 
`self.session.puts` inherits from `requests.exceptions.RequestException` and are handled in the next two conditional statements. Should that be changed?

Reply:
We should catch `requests.exceptions.RequestException` specifically, I think. The issue here isn't the normal exceptions (which generally don't generate exceptions unless you ask them to), but more exotic things like network timeouts or sockets closing on you.